### PR TITLE
add search profiles by ids

### LINKED ISSUE #16: Search profiles by ids
Add new **POST** endpoint `/profiles/search` with **ids** body parameter. Should return **HTTP 200** with list of profiles (the same model as in get by id endpoint)

Input example:
```json
{
    "ids": ["xxx", "fff", "ccc"]
}
```

Output example:
```json
[
    { "id": "xxx", "firstName: "xxxxxx", ..... },
    { "id": "fff", "firstName: "yyyyyy", ..... },
]
```

### PR DESCRIPTION
**Pull request by Sekir0**

### CODE REVIEW

File: GhostNetwork.Profiles.Api/Program.cs [hunk 1]
```diff
@@ -1,11 +1,8 @@
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
+using MongoDB.Bson;
```
Comment (volodymyr-borodin) [inline GhostNetwork.Profiles.Api/Program.cs]: Useless using

File: GhostNetwork.Profiles.Api/Controllers/ProfilesController.cs [hunk 2]
```diff
@@ -18,6 +19,20 @@ public ProfilesController(IProfileService profileService)
...
         }
 
+        /// <summary>
+        /// Search profiles by ids
+        /// </summary>
+        /// <param name="model">Profile ids</param>
+        /// <response code="200">Returns profile</response>
+        [HttpPost("search")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult<IEnumerable<Profile>>> SearchByIdsAsync([FromBody] ProfilesQueryModel model)
```
Comment (volodymyr-borodin) [inline GhostNetwork.Profiles.Api/Controllers/ProfilesController.cs:29]: What is the reason for this endpoint?

**Review by volodymyr-borodin — CHANGES_REQUESTED** [review]

File: GhostNetwork.Profiles.Api/Controllers/ProfilesController.cs (hunk 2 above)
Comment (Sekir0) [reply GhostNetwork.Profiles.Api/Controllers/ProfilesController.cs:29]: https://github.com/ghosts-network/profiles/issues/16
Or I'm didn't correct understand task?

Comment (volodymyr-borodin) [reply GhostNetwork.Profiles.Api/Controllers/ProfilesController.cs:29]: Got it, my mistake. Forgot about issue

File: GhostNetwork.Profiles.Api/Program.cs (hunk 1 above)
Comment (Sekir0) [reply GhostNetwork.Profiles.Api/Program.cs]: Done

File: Infrastructure/GhostNetwork.Profiles.MongoDb/MongoProfileStorage.cs [hunk 3]
```diff
@@ -13,6 +15,17 @@ public MongoProfileStorage(MongoDbContext context)
             this.context = context;
         }
 
+        public async Task<IEnumerable<Profile>> SearchByIdsAsync(IEnumerable<Guid> ids)
+        {
+            var filter = Builders<ProfileEntity>.Filter.In(x => x.Id, ids);
+            
+            var profiles = await context.Profiles.Aggregate()
+                .Match(filter)
+                .ToListAsync();
```
Comment (volodymyr-borodin) [inline Infrastructure/GhostNetwork.Profiles.MongoDb/MongoProfileStorage.cs]:
```suggestion
            var profiles = await context.Profiles
                .Find(filter)
                .ToListAsync();
```

Comment (volodymyr-borodin) [reply Infrastructure/GhostNetwork.Profiles.MongoDb/MongoProfileStorage.cs]: Aggregate for simple Find is overkill

**Pull request merged**